sormas-backend/src/main/resources/sql/sormas_schema.sql

```diff
@@ -12880,4 +12880,38 @@ ALTER TABLE customizableenumvalue_history ADD COLUMN active boolean;
 
 INSERT INTO schema_version (version_number, comment) VALUES (539, 'Add active column to customizable enum values #12804');
 
+-- 2024-02-14 Update environment sample deletion dependencies #12887
+DO
+$$
+    DECLARE
+        user_role_id BIGINT;
+    BEGIN
+        FOR user_role_id IN SELECT DISTINCT(ur.id)
+                            FROM userroles ur
+                                     JOIN userroles_userrights urur ON ur.id = urur.userrole_id
+                            WHERE urur.userright = 'ENVIRONMENT_DELETE'
+                            AND urur.userright != 'ENVIRONMENT_SAMPLE_DELETE'
```

Comment:
the above "not equal" statement has no impact on the query results. That is why if there is at least one user who has both rights the insert below will fail with a duplicate error:

Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "unq_userroles_userrights_0"
  Detail: Key (userrole_id, userright)=(142146, ENVIRONMENT_SAMPLE_DELETE) already exists.
  Where: SQL statement "INSERT INTO userroles_userrights (userrole_id, userright)
                VALUES (user_role_id, 'ENVIRONMENT_SAMPLE_DELETE')"
PL/pgSQL function inline_code_block line 11 at SQL statement